src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -173,6 +175,46 @@ protected AbortException abortUnrecognizedArgument(String argument) {
                 "truffleruby: invalid option " + argument + "  (Use --help for usage instructions.)");
     }
 
+    @Override
+    protected boolean runLauncherAction() {
+        if (System.console() == null) {
+            return super.runLauncherAction();
+        }
+
+        String pager = System.getenv("RUBY_PAGER");
+        if (pager == null) {
+            pager = System.getenv("PAGER");
+
+            if (pager == null) {
+                return super.runLauncherAction();
+            }
+        }
+
+        pager = pager.strip();
```

Comment:
Maybe I can extract it to a separate private method, like `getPagerEnv` which will return `null` or `String` ...? 🤔